doc/nimgrep.rst

```diff
@@ -22,29 +25,44 @@ Compile nimgrep with the command::
 
   nim c -d:release tools/nimgrep.nim
 
-And copy the executable somewhere in your ``$PATH``.
+And copy the executable somewhere in your `$PATH`.
 
 
 Command line switches
 =====================
 
-Usage:
-  nimgrep [options] [pattern] [replacement] (file/directory)*
-Options:
-  --find, -f          find the pattern (default)
-  --replace, -r       replace the pattern
-  --peg               pattern is a peg
-  --re                pattern is a regular expression (default); extended
-                      syntax for the regular expression is always turned on
-  --recursive         process directories recursively
-  --confirm           confirm each occurrence/replacement; there is a chance
-                      to abort any time without touching the file
-  --stdin             read pattern from stdin (to avoid the shell's confusing
-                      quoting rules)
-  --word, -w          the match should have word boundaries (buggy for pegs!)
-  --ignoreCase, -i    be case insensitive
-  --ignoreStyle, -y   be style insensitive
-  --ext:EX1|EX2|...   only search the files with the given extension(s)
-  --verbose           be verbose: list every processed file
-  --help, -h          shows this help
-  --version, -v       shows the version
+.. include:: nimgrep_cmdline.txt
+
+Examples
+========
+
+All examples below use default PCRE Regex patterns:
+
++ To search recursively in Nim files using style-insensitive identifiers::
+
+    --recursive --ext:'nim|nims' --ignoreStyle
+    # short: -r --ext:'nim|nims' -y
+
+  .. Note:: we used `'` quotes to avoid special treatment of `|` symbol
+    for shells like Bash
+
++ To exclude version control directories (Git, Mercurial=hg, Subversion=svn)
+  from the search::
+
+    --excludeDir:'^\.git$' --excludeDir:'^\.hg$' --excludeDir:'^\.svn$'
+    # short: --ed:'^\.git$' --ed:'^\.hg$' --ed:'^\.svn$'
+
++ To search only in paths containing the `tests` sub-directory recursively::
+
+    --recursive --includeDir:'(^|/)tests($|/)'
+    # short: -r --id:'(^|/)tests($|/)'
+
+  .. Attention:: note the subtle difference between `--excludeDir` and
+    `--includeDir`: the former is applied to relative directory entries
+    and the latter is applied to the whole paths
+
++ Nimgrep can search multi-line, e.g. to find files containing `import`
+  and then `strutils` use::
+
+    'import(.|\n)*?strutils'
```

Comment:
* can you make this more precise via `\b`:
```suggestion
    'import(.|\n)*?\bstrutils\b'
```
* doesn't that have false positives, eg:
```nim
import foo
# see also strutils
```
?

Reply:
* I want to keep this example simple but still usable
* yes, it will match even if this "strutils" comment is situated somewhere in the end of file. But Regex is not intended for good semantic checking, right? I mean using it from cmdline supposes quick-and-dirty patterns and hence a number of false positives. (I've heard that it's possible to parse pretty complex grammars by Regex **in theory** -- I'm not a CS guy though)

Reply:
ok, but at least mention this in the comment, eg:
```
+ Nimgrep can search multi-line, e.g. to find files containing `import`
  and then `strutils` use this (this can have false positives)::
```